sklearn/datasets/tests/test_common.py

```diff
@@ -41,3 +63,49 @@ def check_as_frame(bunch, fetch_func_partial,
         assert np.all(frame_bunch.data.dtypes == expected_data_dtype)
     if expected_target_dtype is not None:
         assert np.all(frame_bunch.target.dtypes == expected_target_dtype)
+
+
+def _has_network():
+    return bool(os.environ.get("SKLEARN_SKIP_NETWORK_TESTS", False))
+
+
+def _generate_func_supporting_param(param, dataset_type=("load", "fetch")):
+    markers_known_failure = KNOWN_FAILURE.get(param, {})
+    for name, obj in inspect.getmembers(sklearn.datasets):
+        if not inspect.isfunction(obj):
+            continue
+
+        is_dataset_type = any([name.startswith(t) for t in dataset_type])
+        is_support_param = param in inspect.signature(obj).parameters
+        if is_dataset_type and is_support_param:
+            # check if we should skip if we don't have network support
+            marks = [pytest.mark.skipif(
+                condition=name.startswith("fetch") and _has_network(),
+                reason="Skip because fetcher requires internet network",
+            )]
+            marks.append(markers_known_failure.get(name, pytest.mark.basic))
+
+            yield pytest.param(name, obj, marks=marks)
+
+
+@pytest.mark.parametrize(
+    "name, dataset_func", _generate_func_supporting_param("return_X_y")
+)
+def test_common_check_return_X_y(name, dataset_func):
+    bunch = dataset_func()
+    check_return_X_y(bunch, partial(dataset_func))
+
+
+@pytest.mark.parametrize(
+    "name, dataset_func", _generate_func_supporting_param("as_frame")
+)
+def test_common_check_as_frame(name, dataset_func):
+    bunch = dataset_func()
+    check_as_frame(bunch, partial(dataset_func))
```

Comment:
I am curious, what is the use of `partial` here ?

Reply:
Right it does not seem to be useful. I'll remove it